<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1058</width>
    <height>681</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="3">
     <widget class="QLineEdit" name="lineEdit_column">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>50</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>column</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLineEdit" name="lineEdit_row">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>45</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>row</string>
      </property>
     </widget>
    </item>
    <item row="0" column="7">
     <widget class="QLineEdit" name="lineEdit_obstacle">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>50</string>
      </property>
     </widget>
    </item>
    <item row="0" column="6">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>障碍%</string>
      </property>
     </widget>
    </item>
    <item row="0" column="4">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>width</string>
      </property>
     </widget>
    </item>
    <item row="0" column="5">
     <widget class="QLineEdit" name="lineEdit_width">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>80</string>
      </property>
     </widget>
    </item>
    <item row="0" column="9">
     <widget class="QLineEdit" name="lineEdit_time">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>100</string>
      </property>
     </widget>
    </item>
    <item row="0" column="11">
     <widget class="QComboBox" name="comboBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <item>
       <property name="text">
        <string>曼哈顿距离</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>对角线距离</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>欧几里得距离</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>平方欧几里得距离</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Dijkstra</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Breaking ties</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="0" column="13">
     <widget class="QPushButton" name="pushButton_gen">
      <property name="text">
       <string>gennerate</string>
      </property>
     </widget>
    </item>
    <item row="0" column="12">
     <widget class="QPushButton" name="pushButton_clear">
      <property name="text">
       <string>clear</string>
      </property>
     </widget>
    </item>
    <item row="0" column="10">
     <widget class="QLabel" name="label_6">
      <property name="text">
       <string>启发函数</string>
      </property>
     </widget>
    </item>
    <item row="0" column="8">
     <widget class="QLabel" name="label_5">
      <property name="text">
       <string>time</string>
      </property>
     </widget>
    </item>
    <item row="0" column="14">
     <widget class="QPushButton" name="pushButton_clearPath">
      <property name="text">
       <string>clear path</string>
      </property>
     </widget>
    </item>
    <item row="0" column="15">
     <widget class="QPushButton" name="pushButton_startFinding">
      <property name="text">
       <string>start finding</string>
      </property>
     </widget>
    </item>
    <item row="3" column="13">
     <widget class="QPushButton" name="pushButton_clearLog">
      <property name="text">
       <string>clear log</string>
      </property>
     </widget>
    </item>
    <item row="1" column="13" rowspan="2" colspan="3">
     <widget class="QTextEdit" name="textEdit"/>
    </item>
    <item row="1" column="0" rowspan="3" colspan="13">
     <widget class="GraphicsView" name="graphicsView"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1058</width>
     <height>17</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>GraphicsView.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
